Framework/DataHandling/src/SaveSESANS.cpp

```diff
@@ -0,0 +1,195 @@
+#include "MantidDataHandling/SaveSESANS.h"
+
+#include "MantidAPI/FileProperty.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/Sample.h"
+#include "MantidAPI/WorkspaceProperty.h"
+#include "MantidKernel/make_unique.h"
+
+#include <algorithm>
+#include <cmath>
+#include <fstream>
+#include <iomanip>
+#include <string>
+
+namespace Mantid {
+namespace DataHandling {
+
+// Register the algorithm with the AlgorithmFactory
+DECLARE_ALGORITHM(SaveSESANS)
+
+/// Get the algorithm's name
+const std::string SaveSESANS::name() const { return "SaveSESANS"; }
+
+/// Get a summary of the algorithm
+const std::string SaveSESANS::summary() const {
+  return "Save a file using the SESANS format";
+}
+
+/// Get the version number of the algorithm
+int SaveSESANS::version() const { return 1; }
+
+/// Get the algorithm's category
+const std::string SaveSESANS::category() const { return "DataHandling\\Text"; }
+
+/**
+ * Initialise the algorithm
+ */
+void SaveSESANS::init() {
+  declareProperty(Kernel::make_unique<API::WorkspaceProperty<>>(
+                      "InputWorkspace", "", Kernel::Direction::Input),
+                  "The name of the workspace to save");
+  declareProperty(Kernel::make_unique<API::FileProperty>(
+                      "Filename", "", API::FileProperty::Save, fileExtensions),
+                  "The name to use when saving the file");
+
+  // TODO : find out good descriptions (and validators) for these properties
+  declareProperty("ThetaZMax", -1.0, Kernel::Direction::Input);
+  declareProperty("ThetaZMax_unit", "radians", Kernel::Direction::Input);
+  declareProperty("ThetaYMax", -1.0, Kernel::Direction::Input);
+  declareProperty("ThetaYMax_unit", "radians", Kernel::Direction::Input);
+  declareProperty("EchoConstant", -1.0, Kernel::Direction::Input);
+
+  declareProperty<std::string>("Orientation", "",
```

Comment:
You could use a StringListValidator and give them the option of choosing. This might make it easier to control the input